pkg/protocol/engine/accounts/accountsledger/manager.go

```diff
@@ -404,8 +412,18 @@ func (m *Manager) commitAccountTree(index iotago.SlotIndex, accountDiffChanges m
 		}
 
 		if diffChange.BICChange != 0 || !exists {
-			// TODO: this needs to be decayed for (index - prevIndex) because update index is changed so it's impossible to use new decay
-			// //
+			// decay the credits to the current slot if the account exists
+			if exists {
+				decayedPreviousCredits, err := m.apiProvider.APIForSlot(index).ManaDecayProvider().StoredManaWithDecay(iotago.Mana(accountData.Credits.Value), accountData.Credits.UpdateTime, index)
```

Comment:
I propose that we rename the ManaDecayProvider function as follows.
`StoredManaWithDecay` -> `ManaWithDecay`
`PotentialManaWithDecay` -> `ManaGenerationWithDecay`

Reply:
Done it in https://github.com/iotaledger/iota.go/pull/484